### PR TITLE
add chaincode endpoint with integration test

### PR DESCRIPTION
**Pull request by pallabez**
closes https://github.com/Real-Dev-Squad/website-backend/issues/473

### CODE REVIEW

File: controllers/users.js [hunk 1]
```diff
@@ -143,11 +144,31 @@ const postUserPicture = async (req, res) => {
...
+ */
+const updateChaincode = async (req, res) => {
+  try {
+    const key = uuid.v4()
+    return res.json({
+      message: 'Generated New Chaincode!!',
+      chaincode: key
+    })
+  } catch (error) {
+    logger.error(`Error while updating user: ${error}`)
```
Comment (lakshayman) [inline controllers/users.js:161]:
```suggestion
    logger.error(`Error while updating user's chaincode: ${error}`)
```

**Review by lakshayman — DISMISSED** [review]
Else Looks fine to me

**Review by lakshayman — APPROVED** [review]
Great! Good Work

**Comment by shubham-y** [comment]
@pallabez Are we updating the newly generated chain code to the user document?

**Comment by pallabez** [comment]
> @pallabez Are we updating the newly generated chain code to the user document?

No, we are going to store the newly generated chain code in a different document that hasn't been created yet.

**Comment by shubham-y** [comment]
> > @pallabez Are we updating the newly generated chain code to the user document?
> 
> No, we are going to store the newly generated chain code in a different document that hasn't been created yet.
So for now, shouldn't we create a GET chain code endpoint as we are updating anything. What do you think?

**Comment by pallabez** [comment]
> > > @pallabez Are we updating the newly generated chain code to the user document?
> > 
> > 
> > No, we are going to store the newly generated chain code in a different document that hasn't been created yet.
> 
> So for now, shouldn't we create a GET chain code endpoint as we are updating anything. What do you think?

Seems reasonable. I'll ask Lakshay about it.

**Review by lakshayman — APPROVED** [review]
Looks Good!

File: controllers/users.js [hunk 2]
```diff
@@ -143,11 +144,31 @@ const postUserPicture = async (req, res) => {
...
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+const getChaincode = (req, res) => {
+  try {
+    const key = uuid.v4()
+    return res.json({
+      message: 'Generated New Chaincode!!',
+      chaincode: key
```
Comment (ankurnarkhede) [inline controllers/users.js:158]: How are we associating the code with a specific user? This controller is just generating a UUID v4, that we could have generated on the UI itself if we dont want to associate it with a user... 🤔

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: controllers/users.js (hunk 2 above)
Comment (ankushdharkar) [reply controllers/users.js:158]: Good catch @ankurnarkhede. We should store the chaincode in firestore and read the document id as a key (helps prevent potential collisions also).

@pallabez Can you add that as a comment please, so the next person (maybe you?) can remember to make the appropriate changes. Attach the ticket link in the comments please.

Comment (pallabez) [reply controllers/users.js:158]: Got it 👍

**Pull request closed without merging**